packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/styles.ts

```diff
@@ -267,7 +267,7 @@ export function getStylesConfig(wco: WebpackConfigOptions) {
             options: {
               ident: buildOptions.extractCss ? 'extracted' : 'embedded',
               plugins: postcssPluginCreator,
-              sourceMap: cssSourceMap
+              sourceMap: cssSourceMap ? 'inline' : false
```

Comment:
I'm assuming based on the linked issue that you meant to change the source map configuration for component styles located [here](https://github.com/ajspera/angular-cli/blob/4e569e575c0affdc8b015559fecf8fa9748bde6e/packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/styles.ts#L247) and not the global styles config.

Reply:
Actually both are affected by the problem, but I had missed that there were 2 instances of postcss-loader. With each using 'inline' instead of a boolean, both the global maps and component maps are now working. I updated the commit and the commit message to reflect that.

Reply:
If that is the case then the comment in this section is very relevant.  `style-loader` 1.0 (unreleased) automatically handles the source maps in this regard.  As a stopgap this appears to be an acceptable solution.  However, if the global stylesheets are extracted, they should not contain inline source maps.  The conditional in this case should set the value to `true` if source maps and extraction is enabled; `inline` if only source maps are enabled; and `false` otherwise.

Reply:
I had thought about that, but when I tried `ng build --source-map --extract-css` without the inline sourcemaps, they did not work. The chrome inspector just referenced the generated css files. Looking a bit further, it does generate a map file for the styles, but there are no mappings to the scss files in it.
#### dist/styles.css.map
```
{"version":3,"sources":[],"names":[],"mappings":"","file":"styles.css","sourceRoot":""}
```
There may be another bug happening on top of this, although the always inline will resolve it as is until it's found. I suppose that's up to you whether you want extracted sourcemaps broken until it's found or supplement it as is (always inline if true) until then.

Reply:
Extracted stylesheets should have working source maps.   If they are not working that would be a separate issue.  Altering the behavior for extracted souremaps in this way would be require a major version bump as end users may not realize the source maps are embedded resulting in the source being pushed unknowingly in production builds.

Reply:
It will still only push source maps if --source-map is explicitly set, but I suppose your point may be relevant if people are building source maps and then manually pulling the map files from their build as part of a deployment process which I would consider unorthodox. I do agree that is a separate issue, but its also related to the referenced issue since it isn't completely resolved until the extracted scss source maps also work.

Reply:
If you are willing to make the suggested change, we should be able to roll this into the next minor.

Reply:
Cool, I should be able to get that in today. Just to confirm, the component styles are always injected inline via js even with extract-css, so the change just goes on the global styles, right?

Reply:
yes.

Reply:
The change is up.